app/controllers/activities_controller.rb

```diff
@@ -16,5 +16,6 @@ def show
 
     def set_activity
       @activity = Activity.find(params[:id])
+      @media_content = @activity.media_content_id.present? ? MediaContent.find(@activity.media_content_id) : nil
```

Comment:
If the associations are set in the model you can just do:

`@media_content = @activity.media_content`

Will be nil if there's no record associated.